### ISSUE TITLE
`pod_spec` field should be validated before being passed to K8S

### ISSUE THREAD

**Issue by gabrielgrant** [pachyderm/pachyderm#3412]
Right now it seems the `pod_spec` field isn't validated at all before being passed off to k8s. The result is that even brain-dead-simple problems (like JSON syntax errors) get swallowed as part of the k8s resource creation process

See @mdaniel's report for example: https://github.com/pachyderm/pachyderm/issues/3059#issuecomment-458654384

Would it be possible for us to just import and use the k8s protobuf definition of a pod spec and include that as a normal, parsed (and, therefore, type-checked) field in the PipelineInfo protobuf?

**Comment by mdaniel** [comment]
> If you have a minute, can you please share the output of kubectl get all and kubectl describe po/<bug3059-pod-name>?

_I'm relocated that thread over here, since its seems more relevant to this issue_

As you might suspect, there is no Pod created because the RC was never created because pachd is banging its head against the RC creation process:

```
2019-01-29T23:34:29Z INFO pfs.API.GetFile {"duration":0.004561045,"request":{"file":{"commit":{"repo":{"name":"__spec__"},"id":"a4fd2e26771e43c6b69ceb4ca62fc36b"},"path":"spec"}}} []
time="2019-01-29T23:34:29Z" level=info msg="PPS master: creating/updating workers for pipeline bug3059"
2019-01-29T23:34:29Z INFO enterprise.API.GetState {"request":{}} []
2019-01-29T23:34:29Z INFO enterprise.API.GetState {"duration":0.000002325,"request":{},"response":{}} []
time="2019-01-29T23:34:29Z" level=error msg="error creating workers for pipeline bug3059: invalid character '{' after array element; retrying in 530.666954ms"
2019-01-29T23:34:29Z INFO enterprise.API.GetState {"request":{}} []
2019-01-29T23:34:29Z INFO enterprise.API.GetState {"duration":0.000002031,"request":{},"response":{}} []
time="2019-01-29T23:34:29Z" level=error msg="error creating workers for pipeline bug3059: invalid character '{' after array element; retrying in 792.533877ms"
2019-01-29T23:34:30Z INFO enterprise.API.GetState {"request":{}} []
2019-01-29T23:34:30Z INFO enterprise.API.GetState {"duration":0.000002088,"request":{},"response":{}} []
time="2019-01-29T23:34:30Z" level=info msg="moving pipeline bug3059 to PIPELINE_STARTING"
time="2019-01-29T23:34:30Z" level=error msg="master: error running the master process: invalid character '{' after array element; retrying in 8.686887543s"
```

**Comment by gabrielgrant** [comment]
Oh, yes, of course, that makes sense. We just spent some time digging into the issue of having a `reason` field set, but not transitioning into an error state. Basically, it seems what's going on is that this error is reported in the same way as a number of transient k8s errors.

Our default behavior is to retry any operation that _could_ be a transient failure (ie that we haven't hard-coded to be fatal), rather than bailing on every network hiccup. What we've just discussed, though, is being more explicit about distinguishing states that are "good/expected" vs "possibly good/attempting recovery" (today those are mostly conflated)
